src/daft-logical-plan/src/optimization/rules/unnest_subquery.rs

```diff
@@ -0,0 +1,779 @@
+            LogicalPlan::Filter(Filter {
+                input, predicate, ..
+            }) => {
+                let mut subqueries = HashSet::new();
+
+                let new_predicates = split_conjuction(predicate)
+                    .into_iter()
+                    .filter(|expr| {
+                        match expr.as_ref() {
+                            Expr::InSubquery(in_expr, subquery) => {
+                                subqueries.insert(PredicateSubquery { subquery: subquery.clone(), in_expr: Some(in_expr.clone()), join_type: JoinType::Semi });
+                                false
+                            }
+                            Expr::Exists(subquery) => {
+                                subqueries.insert(PredicateSubquery { subquery: subquery.clone(), in_expr: None, join_type: JoinType::Semi });
+                                false
+                            }
+                            Expr::Not(e) => {
+                                match e.as_ref() {
+                                    Expr::InSubquery(in_expr, subquery) => {
+                                        subqueries.insert(PredicateSubquery { subquery: subquery.clone(), in_expr: Some(in_expr.clone()), join_type: JoinType::Anti });
+                                        false
+                                    }
+                                    Expr::Exists(subquery) => {
+                                        subqueries.insert(PredicateSubquery { subquery: subquery.clone(), in_expr: None, join_type: JoinType::Anti });
+                                        false
+                                    }
+                                    _ => true
+                                }
+                            }
+                            _ => true
+                        }
+                    })
+                    .cloned()
+                    .collect::<Vec<_>>();
+
+                if subqueries.is_empty() {
+                    return Ok(Transformed::no(node));
+                }
+
+                let new_input = subqueries.into_iter().try_fold(input.clone(), |curr_input, PredicateSubquery { subquery, in_expr, join_type }| {
+                    let subquery_plan = downcast_subquery(&subquery);
+
+                    let (decorrelated_subquery, mut subquery_on, mut input_on) =
+                        pull_up_correlated_cols(subquery_plan.clone())?;
+
+                    if let Some(in_expr) = in_expr {
+                        let subquery_col_names = subquery_plan.schema().names();
+                        let [output_col] = subquery_col_names.as_slice() else {
+                            return Err(DaftError::ValueError(format!("Expected IN subquery to have one output column, received: {}", subquery_col_names.len())));
+                        };
+
+                        input_on.push(in_expr);
+                        subquery_on.push(col(output_col.as_str()));
+                    }
```

Comment:
nit: we can avoid a clone here 
```suggestion
                    let subquery_plan = downcast_subquery(&subquery);
                    let subquery_schema = subquery_plan.schema();

                    let (decorrelated_subquery, mut subquery_on, mut input_on) =
                        pull_up_correlated_cols(subquery_plan)?;

                    if let Some(in_expr) = in_expr {
                        let subquery_col_names = subquery_schema.names();
                        let [output_col] = subquery_col_names.as_slice() else {
                            return Err(DaftError::ValueError(format!("Expected IN subquery to have one output column, received: {}", subquery_col_names.len())));
                        };

                        input_on.push(in_expr);
                        subquery_on.push(col(output_col.as_str()));
                    }
```